engine/util/src/main/scala/pl/touk/nussknacker/engine/util/cache/DefaultCache.scala

```diff
@@ -0,0 +1,79 @@
+package pl.touk.nussknacker.engine.util.cache
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import com.github.benmanes.caffeine.cache.stats.ConcurrentStatsCounter
+import scalacache.Entry
+
+import scala.concurrent.duration.Duration
+
+/**
+  * @param maximumSize the maximum elements number can contain cache
+  * @param expireAfterAccess the expiration time from last action (read / write)
+  * @tparam T
+  */
+class DefaultCache[T](maximumSize: Long, expireAfterAccess: Option[Duration], registerStats: Boolean) extends Cache[T] {
+
+  import scalacache.caffeine._
+  import scalacache.modes.sync._
+
+  import scala.concurrent.duration._
+
+  private val stats = new ConcurrentStatsCounter
```

Comment:
Why we don't just use .recordStats() as in: https://github.com/ben-manes/caffeine/wiki/Statistics?

Reply:
It's more complicated.. Caffaine doesn't support ttl per key, we use only expierAfterAccess for all keys.

In our case we use two types of expiration:
- expierAfterAccess for all keys
- ttl for key from scalacache: https://github.com/cb372/scalacache/blob/master/modules/caffeine/src/main/scala/scalacache/caffeine/CaffeineCache.scala#L25

It's more complex detecting and cleanup cache. For that I have to use own stats counter.

Reply:
So maybe we should split this cache instance into two caches (one for parsed schemas) and one for (latest -> schema)? I don't like this idea of own counters handling - take a look at my notice about "cirtical section". There is a log of things that can go wrong.

Reply:
I've removed scalacache - there is only Caffeine.